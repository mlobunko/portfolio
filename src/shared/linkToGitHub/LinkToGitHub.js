import React from "react";
import PropTypes from "prop-types";

import { StyledLink } from "./styles";

export const LinkToGitHub = ({ link, nameOfProject, textColor }) => (
  <StyledLink href={link} nameOfProject={nameOfProject} textColor={textColor}>
    Visit GitHub
  </StyledLink>
);

LinkToGitHub.displayName = "LinkToGithub";

export default LinkToGitHub;

LinkToGitHub.propTypes = {
  link: PropTypes.string.isRequired,
  nameOfProject: PropTypes.string.isRequired,
  textColor: PropTypes.string
};
