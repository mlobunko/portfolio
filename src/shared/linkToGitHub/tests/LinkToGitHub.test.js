import React from "react";
import { shallow } from "enzyme";

import { LinkToGitHub } from "../LinkToGitHub";

describe("render", () => {
  it("renders correctly", () => {
    const props = {
      link: "123",
      nameOfProject: "family",
      textColor: "black"
    };
    const wrapper = shallow(<LinkToGitHub {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
