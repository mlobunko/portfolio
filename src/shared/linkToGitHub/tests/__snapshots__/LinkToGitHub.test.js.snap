// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<StyledLink
  href="123"
  nameOfProject="family"
  textColor="black"
>
  Visit GitHub
</StyledLink>
`;
